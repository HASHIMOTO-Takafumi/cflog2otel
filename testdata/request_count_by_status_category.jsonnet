local cel = std.native('cel');

{
  otel: {
    endpoint: 'http://localhost:4317/',
    gzip: true,
  },
  resource_attributes: [
    {
      key: 'service.name',
      value: 'Amazon CloudFront',
    },
    {
      key: 'aws.cloudfront.distribution_id',
      value: cel('cloudfront.distributionId'),
    },
  ],
  scope: {
    name: 'test',
  },
  metrics: [
    {
      name: 'http.server.requests',
      description: 'The number of HTTP requests',
      type: 'Counter',
      attributes: [
        {
          key: 'http.status_code',
          value: cel('log.scStatusCategory'),
        },
      ],
    },
  ],
}
