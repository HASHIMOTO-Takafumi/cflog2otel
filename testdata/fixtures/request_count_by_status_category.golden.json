{
  "Resource": [
    {
      "Key": "aws.cloudfront.distribution_id",
      "Value": {
        "Type": "STRING",
        "Value": "EMLARXS9EXAMPLE"
      }
    },
    {
      "Key": "service.name",
      "Value": {
        "Type": "STRING",
        "Value": "Amazon CloudFront"
      }
    }
  ],
  "ScopeMetrics": [
    {
      "Scope": {
        "Name": "test",
        "Version": "1.0.0",
        "SchemaURL": "https://example.com/schemas/1.0.0"
      },
      "Metrics": [
        {
          "Name": "http.server.requests",
          "Description": "The number of HTTP requests",
          "Unit": "",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "2xx"
                    }
                  }
                ],
                "StartTime": "2019-12-04T21:02:00Z",
                "Time": "2019-12-04T21:02:59.999999999Z",
                "Value": 3
              },
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-13T22:36:00Z",
                "Time": "2019-12-13T22:36:59.999999999Z",
                "Value": 2
              },
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-13T22:37:00Z",
                "Time": "2019-12-13T22:37:59.999999999Z",
                "Value": 1
              }
            ],
            "Temporality": "DeltaTemporality",
            "IsMonotonic": true
          }
        }
      ]
    }
  ]
}